QuantConnect.GDAXBrokerage/GDAXBrokerage.cs

```diff
@@ -178,27 +209,22 @@ public override bool UpdateOrder(Order order)
         /// <returns></returns>
         public override bool CancelOrder(Order order)
         {
-            var success = new List<bool>();
+            var cancelOrder = _coinbaseApi.CancelOrders(order.BrokerId);
 
-            foreach (var id in order.BrokerId)
+            if (!cancelOrder.Success)
             {
-                var req = new RestRequest("/orders/" + id, Method.DELETE);
-                GetAuthenticationToken(req);
-                var response = ExecuteRestRequest(req, GdaxEndpointType.Private);
-                success.Add(response.StatusCode == HttpStatusCode.OK);
-                if (response.StatusCode == HttpStatusCode.OK)
-                {
-                    OnOrderEvent(new OrderEvent(order,
-                        DateTime.UtcNow,
-                        OrderFee.Zero,
-                        "GDAX Order Event") { Status = OrderStatus.Canceled });
-
-                    PendingOrder orderRemoved;
-                    _pendingOrders.TryRemove(id, out orderRemoved);
-                }
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Error, "CancelOrder",
+                    $"Coinbase has not canceled order, error: {cancelOrder.FailureReason}"));
+                return false;
             }
 
-            return success.All(a => a);
+            // TODO: Why did we use this ?
```

Comment:
Could look at `_pendingOrders` and `CachedOrderIds` usages to find out how they are used. If not use I guess they could be removed

Reply:
Removed.